index.js

```diff
@@ -0,0 +1,369 @@
+
+                if (!signingKey) {
+                    throw new Error('No valid key exists to support this time range');
+                }
+
+                return Jwt.sign({
+                    entity: parsedConfig.entity,
+                    scope: parsedConfig.scope
+                }, signingKey.private, {
+                    algorithm: signingKey.algorithm,
+                    expiresIn: parsedConfig.time,
+                    notBefore: 0,
+                    issuer: authConfig.issuer,
+                    jwtid: guid(),
+                    keyid: signingKeyId,
+                    subject: parsedConfig.subject
+                });
+            },
+
+            /**
+             * List only the active keys (no larger than maxAge)
+             * @method availableKeys
+             * @return {Object}      KeyId => public, expire, algorithm
+             */
+            availableKeys: () => {
+                const nowDate = Date.now() / 1000;
+                const maxDate = nowDate + (ms(authConfig.maxAge) / 1000);
+                const output = {};
+
+                expireIds.some((keyId) => {
+                    if (nowDate > keys[keyId].expires) {
+                        return false;
+                    }
+
+                    output[keyId] = {
+                        public: keys[keyId].public,
+                        algorithm: keys[keyId].algorithm,
+                        expires: keys[keyId].expires
+                    };
+
+                    if (maxDate > keys[keyId].expires) {
+                        return false;
+                    }
+
+                    return true;
+                });
+
+                return output;
+            }
+        }
+    };
+};
+
+/**
+ * Creates a Remote auth service
+ * @method loadRemote
+ * @param  {Object}  authConfig           Input for setting up a remote version
+ * @param  {String}  authConfig.remoteUri URI of remote server
+ * @param  {String}  authConfig.timeout   How long to wait for API response
+ * @return {Object}                       Contains keys, apis, etc.
+ */
+internals.loadRemote = (authConfig) => {
+    const mutex = semaphore(1);
+    let keys = {};
+
+    return {
+        /**
+         * Load the key data (from the remote server if needed)
+         * @method getKey
+         * @param  {String}   keyId Key you are looking for
+         * @param  {Function} next  Function to call when done (error, key)
+         */
+        getKey: (keyId, next) => {
+            const nextSem = (err, result) => {
+                mutex.leave();
+                next(err, result);
+            };
+
+            // Prevent multiple calls to the remote URI
+            mutex.take(() => {
+                if (keys[keyId]) {
+                    return nextSem(null, keys[keyId]);
+                }
+
+                return request({
+                    url: authConfig.remoteUri,
+                    headers: {
```

Comment:
what is authentication model in this request to fetch remote key - between remote key server and api server ?

Reply:
No authentication.  These are public keys.  I could expose the request options, so that users _could_ do something like client TLS.